wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -54,15 +54,21 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
             newWallet
           }
       }
-    res.onComplete(failure =>
+    resF.onComplete(failure =>
       signalBlockProcessingCompletion(block.blockHeader.hash, failure))
-    res.foreach(_ =>
+    resF.foreach(_ =>
```

Comment:
I think these log messages should be done after the `updateSyncHeight` call is successful, so that the logs are atomic. 

As currently stands, you can block processing work, but updating sync height failing. From looking at the logs I would think everything succeeded.

This does open a broader question of what to do in the case of sync height failing to update, but block processing works. Can anything nasty happen there?